tests/examples/simple.cpp

```diff
@@ -20,6 +20,39 @@ struct Error final {
   std::string what;
 };
 
+template <typename T> struct immovable_Value final {
+  T value;
+
+  constexpr immovable_Value(T &&v) noexcept : value(v) {}
+  constexpr auto operator==(immovable_Value const &) const noexcept -> bool
+      = default;
+
+  constexpr immovable_Value(immovable_Value const &) = delete;
+  constexpr immovable_Value(immovable_Value &&) = delete;
+  constexpr immovable_Value &operator=(immovable_Value const &) = delete;
+  constexpr immovable_Value &operator=(immovable_Value &&) = delete;
+};
+template <typename T> immovable_Value(T &&) -> immovable_Value<T>;
+
+template <typename Fn> struct immovable_Fn final {
+  Fn fn;
+
+  constexpr immovable_Fn(Fn &&fn) noexcept : fn(fn) {}
+  constexpr auto operator()(auto &&...args) const noexcept -> decltype(auto)
+    requires requires { fn(args...); }
+  {
+    return fn(std::forward<decltype(args)>(args)...);
+  }
+
+  constexpr immovable_Fn(immovable_Fn const &) = delete;
+  constexpr immovable_Fn(immovable_Fn &&) = delete;
+  constexpr immovable_Fn &operator=(immovable_Fn const &) = delete;
+  constexpr immovable_Fn &operator=(immovable_Fn &&) = delete;
+};
+template <typename Fn> immovable_Fn(Fn &&) -> immovable_Fn<Fn>;
+
+template <typename T> struct incomplete;
+
 TEST_CASE(
     "Demo expected",
     "[expected][and_then][transform_error][transform][inspect][recover][fail]")
```

Comment:
Perhaps worth adding the `[immovable]` tag as well?

Reply:
will do